recipes/zlib/1.2.11/conanfile.py

```diff
@@ -65,7 +65,9 @@ def _build_zlib_autotools(self):
         else:
             make_target = "libz.a"
 
-        env = {"CC": "clang", "CXX": "clang++"} if "clang" in str(self.settings.compiler) else {}
+        env = {}
+        if "clang" in str(self.settings.compiler) and tools.get_env("CC") is None and tools.get_env("CXX") is None:
```

Comment:
probably, we shouldn't touch it here at all, but it's still an improvement on its own